File: apps/api/app/schemas/onboarding.py
from typing import Any, Literal

from pydantic import BaseModel, Field


TargetModule = Literal["contract", "dispatch", "decision", "procurement"]
BlockReason = Literal[
    "not_searchable",
    "module_exists",
    "manual_metadata",
    "unmapped_document_type",
    "low_confidence",
]


class OnboardingSuggestionResponse(BaseModel):
    document_id: str
    eligible: bool
    block_reason: BlockReason | None = None
    needs_metadata_review: bool = False
    suggested_business_type: str | None = None
    business_type_confidence: float | None = Field(default=None, ge=0.0, le=1.0)
    target_module: TargetModule | None = None
    module_confidence: float | None = Field(default=None, ge=0.0, le=1.0)
    module_kind: str | None = None
    reasons: list[str] = Field(default_factory=list)
    suggested_module_fields: dict[str, Any] = Field(default_factory=dict)

File: apps/web/types/onboarding.ts
export type TargetModule = 'contract' | 'dispatch' | 'decision' | 'procurement'

export type OnboardingBlockReason =
  | 'not_searchable'
  | 'module_exists'
  | 'manual_metadata'
  | 'unmapped_document_type'
  | 'low_confidence'

export interface OnboardingSuggestion {
  document_id: string
  eligible: boolean
  block_reason: OnboardingBlockReason | null
  needs_metadata_review: boolean
  suggested_business_type: string | null
  business_type_confidence: number | null
  target_module: TargetModule | null
  module_confidence: number | null
  module_kind: string | null
  reasons: string[]
  suggested_module_fields: Record<string, string | number | boolean | null>
}
